src/gui/EntryTree.cpp

```diff
@@ -626,12 +628,22 @@ void EntryTree::removeEntryRecurse(QTreeWidgetItem* item) {
 
 void LoadPackFileWorker::run(EntryTree* tree, const PackFile& packFile) {
 	int progress = 0;
+    // Neither the tree nor the model uses it's signals as items are being put in.
+    // Firing signals slows the item creation process by a whole 50%! So we disable it.
+    // During the creation process and turn it back on when it's done.
+    tree->blockSignals(true);
```

Comment:
Is it possible to use QSignalBlocker here?

Reply:
As far as I know, blockSignals has identical functionality and I assume that internally QT uses the same code. Why would you want to recreate it?

Reply:
QSignalBlocker is safer since it uses RAII to automatically unblock signals from the given object when it goes out of scope